src/PowerShellEditorServices/Language/TokenOperations.cs

```diff
@@ -21,103 +21,140 @@ internal static class TokenOperations
         private const string RegionKindRegion = "region";
         private const string RegionKindNone = null;
 
-        // Opening tokens for { } and @{ }
-        private static readonly TokenKind[] s_openingBraces = new []
-        {
-            TokenKind.LCurly,
-            TokenKind.AtCurly
-        };
-
-        // Opening tokens for ( ), @( ), $( )
-        private static readonly TokenKind[] s_openingParens = new []
-        {
-            TokenKind.LParen,
-            TokenKind.AtParen,
-            TokenKind.DollarParen
-        };
+        // These regular expressions are used to match lines which mark the start and end of region comment in a PowerShell
+        // script. They are based on the defaults in the VS Code Language Configuration at;
+        // https://github.com/Microsoft/vscode/blob/64186b0a26/extensions/powershell/language-configuration.json#L26-L31
+        static private readonly Regex s_startRegionTextRegex = new Regex(
+           @"^\s*#region\b",
+            RegexOptions.IgnoreCase | RegexOptions.Compiled);
```

Comment:
Do we want `IgnoreCase` here?  ISE only accepts `#region` not `#Region` or `#REGION` or `#rEgIoN`.  Given that the VSCode allows #region and #Region but not any other combination, this regex should drop the `IgnoreCase` and use `@"^\s*#[rR]egion\b"` to match the VSCode defintion.  Same goes for endregion below - `@"^\s*#[eE]nd[rR]egion\b" - which allows unfortunate combo's like `endRegion` and `Endregion`.  Oh well.

Reply:
Nice pickup! Fixed.